meta/SelfEnergies.m

```diff
@@ -618,29 +618,49 @@ therefore not be accessed in the form Glu(gO2).
            Utils`StartProgressBar[Dynamic[k], Length[nPointFunctions]];
            For[k = 1, k <= Length[nPointFunctions], k++,
                Utils`UpdateProgressBar[k, Length[nPointFunctions]];
-               {p,d} = CreateNPointFunction[nPointFunctions[[k]], vertexFunctionNames];
-               prototypes = prototypes <> p;
-               defs = defs <> d;
-               {p,d} = CreateNPointFunctionMatrix[nPointFunctions[[k]]];
-               prototypes = prototypes <> p;
-               defs = defs <> d;
+               {prototype,def} = CreateNPointFunction[nPointFunctions[[k]], vertexFunctionNames];
+               prototypes = prototypes <> prototype;
+               defs = defs <> def;
+               {prototype,def} = CreateNPointFunctionMatrix[nPointFunctions[[k]]];
+               prototypes = prototypes <> prototype;
+               defs = defs <> def;
            ];
            (* create derivatives of Higgs boson self-energies w.r.t. p^2 *)
            If[ValueQ[SARAH`HiggsBoson],
-              derivatives = Cases[nPointFunctions, FSSelfEnergy[SARAH`HiggsBoson | SARAH`HiggsBoson[__], ___]] /. {
-                  FSSelfEnergy -> SelfEnergies`FSSelfEnergyDerivative,
-                  A0[__] -> 0,
-                  B0 -> SelfEnergies`DB0,
-                  G0 -> SelfEnergies`DG0
-              };
+              derivatives = Cases[nPointFunctions, FSSelfEnergy[SARAH`HiggsBoson | SARAH`HiggsBoson[__], ___]];
+
+	      Unprotect[D];
+	      D[SelfEnergies`FSSelfEnergy[particle_, expr_], mom2_] := SelfEnergies`FSSelfEnergyDerivative[particle, D[expr, mom2]];
+	      Derivative[1, 0, 0][B0][p2_, m12_, m22_] := DB0[p2, m12, m22];
+	      Derivative[1, 0, 0][F0][p2_, m12_, m22_] := DF0[p2, m12, m22];
+	      Derivative[1, 0, 0][G0][p2_, m12_, m22_] := DG0[p2, m12, m22];
+	      Protect[D];
+
+	      SetSystemOptions[
```

Comment:
Maybe this `SetSystemOptions[...]` could also be placed outside the function at module scope.